src/test/java/core/basesyntax/converter/impl/DataConverterTest.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax.converter.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class DataConverterTest {
+    private static DataConverter dataConverter;
+
+    @BeforeAll
+    static void beforeAll() {
+        dataConverter = new DataConverter();
+    }
+
+    @Test
+    void converter_fruitData_ShouldReturnWholeList() {
+        List<String> data = getData();
+        List<FruitTransaction> actual = dataConverter.convertToTransaction(data);
+        List<FruitTransaction> expected = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 20),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 100),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 13),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 10),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "apple", 20),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 5),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 50));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void convertToTransaction_invalidNumber_shouldThrowNumberFormatException() {
+        List<String> report = List.of(
+                "type,fruit,quantity",
+                "BUY,apple,invalidNumber");
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(report));
+    }
+
+    @Test
+    void convertToTransaction_emptyList_shouldReturnEmptyList() {
+        List<String> report = new ArrayList<>();
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(report);
+        assertTrue(transactions.isEmpty());
+    }
+
+    private static List<String> getData() {
+        List<String> data = new ArrayList<>();
+        data.add("type,fruit,quantity");
+        data.add("b,banana,20");
+        data.add("b,apple,100");
+        data.add("p,banana,13");
+        data.add("r,apple,10");
+        data.add("p,apple,20");
+        data.add("p,banana,5");
+        data.add("s,banana,50");
+        return data;
+    }
```

Comment:
OPTIONAL: Actually you need to use some mock data in most of tests, so It makes sense to reuse methods that create mock test data to make it more clear, to make this you can create a class (TestObjectMother for example) that will contain all constants and methods that required in your tests

Reply:
Actually ,I know how to use Mockito but I'm unsure that I can use it there. I have seen your recommendations. Thank you.